app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Sessions Helper
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def log_out
+    forget(current_user)
+    session.delete(:user_id)
+    @current_user = nil
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+end
```

Comment:
Files in the `helpers` directory are more specifically `view helpers`
Having these methods defined in the helpers puts a lot of your logic on the views which is not good. 

Move the methods to the corresponding controller. If you need to have a method available between controllers you can define it on the `application_controller`

Reply:
http://codefol.io/posts/where-do-i-put-my-code-in-rails-updated/